### PR TITLE
modernize a variety of code

### PR DESCRIPTION
**Pull request by grimreaper**
- weaken types
- reduce visibility
- prefer lambdas where they simplify code
- delete unused code
- removed useless javadoc comments

(mostly I'm using this as a way to spelunk through the code)

### CODE REVIEW

File: code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java [hunk 1]
```diff
@@ -113,7 +113,7 @@
 	
 	private final JTreeTable companionsTable;
 	private final JEditorPane infoPane;
-	private final JButton loadButton;
+	final JButton loadButton;
```
Comment (cpmeister) [inline code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java]: Why?

Comment (grimreaper) [inline code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java]: mistake - fixed.

File: code/src/java/pcgen/system/ConsoleUIDelegate.java [hunk 2]
```diff
@@ -109,37 +108,25 @@ public void showWarningMessage(String title, String message)
...
-	 */
 	@Override
 	public CustomEquipResult showCustomEquipDialog(CharacterFacade character, 
 		EquipmentBuilderFacade equipBuilder)
 	{
 		return CustomEquipResult.CANCELLED;
 	}
 
-	/**
-	 * {@inheritDoc}
```
Comment (thpr) [inline code/src/java/pcgen/system/ConsoleUIDelegate.java]: Curious why we are deleting these?  It will then trigger an error from our Checkstyle setup.

Comment (grimreaper) [inline code/src/java/pcgen/system/ConsoleUIDelegate.java]: The checkstyle setup should change: similar docs are automatically produced and it adds to the overhead of understanding what a given function is doing (every comment adds a small amount of overhead to parse and maybe ignore).

File: code/src/java/pcgen/rules/context/AbstractListContext.java [hunk 3]
```diff
@@ -60,11 +60,6 @@ void setSourceURI(URI sourceURI)
 		getCommitStrategy().setSourceURI(sourceURI);
 	}
 
-	URI getExtractURI()
```
Comment (thpr) [inline code/src/java/pcgen/rules/context/AbstractListContext.java]: This is necessary for future code (some of which I have in a local branch), so please do not remove.

File: code/src/java/pcgen/rules/persistence/CDOMSubLineLoader.java [hunk 4]
```diff
@@ -111,36 +109,13 @@ public T getCDOMObject()
...
 		return targetPrefix;
 	}
 
-	public Class<T> getLoadedClass()
+	Class<T> getLoadedClass()
 	{
 		return targetClass;
 	}
-
-	public void unloadObject(LoadContext lc, T object, StringBuilder sb)
```
Comment (thpr) [inline code/src/java/pcgen/rules/persistence/CDOMSubLineLoader.java]: Was written for future used of unloading (related to the getExtractURI method above)

File: code/src/java/pcgen/rules/persistence/TokenLibrary.java [hunk 5]
```diff
@@ -559,24 +559,6 @@ protected T grabToken(TokenFamily family, Class<?> cl, String key)
 
 	}
 
-	static class PreTokenIterator
```
Comment (thpr) [inline code/src/java/pcgen/rules/persistence/TokenLibrary.java]: For extract/unload as well

**Pull request merged**